cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -87,6 +88,39 @@ impl LockedManifest {
 
         Ok(PathBuf::from(result.store_path))
     }
+
+    /// Build a container image from a locked manifest
+    /// and write it to a provided sink.
+    ///
+    /// The sink can be e.g. a [File](std::fs::File), [Stdout](std::io::Stdout),
+    /// or an internal buffer.
+    pub fn build_container(
+        &self,
+        pkgdb: &Path,
+        sink: &mut (impl Write + ?Sized),
+    ) -> Result<(), LockedManifestError> {
+        let mut pkgdb_cmd = Command::new(pkgdb);
+        pkgdb_cmd.arg("buildenv").arg(&self.to_string());
+
+        debug!("building container builder with command: {pkgdb_cmd:?}");
+        let result: BuildEnvResult =
+            serde_json::from_value(call_pkgdb(pkgdb_cmd).map_err(LockedManifestError::BuildEnv)?)
+                .map_err(LockedManifestError::ParseBuildEnvOutput)?;
+
+        let container_builder_path = PathBuf::from(result.store_path);
+
+        let mut container_builder_command = Command::new(container_builder_path);
```

Comment:
nonblocking: for the sake of status messages, would it be good to move everything after this to `Containerize::handle()`? So step 1 can be `Building container...` and step 2 can be `Streaming container...`